protocol/x/vault/keeper/orders_test.go

```diff
@@ -23,6 +24,196 @@ const (
 	orderExpirationSeconds             = uint32(5)     // 5 seconds
 )
 
+func TestRefreshAllVaultOrders(t *testing.T) {
+	tests := map[string]struct {
+		// Vault IDs.
+		vaultIds []vaulttypes.VaultId
+		// Total Shares of each vault ID above.
+		totalShares []vaulttypes.NumShares
+	}{
+		"Two Vaults, Both Positive Shares": {
+			vaultIds: []vaulttypes.VaultId{
+				constants.Vault_Clob_0,
+				constants.Vault_Clob_1,
+			},
+			totalShares: []vaulttypes.NumShares{
+				{
+					NumShares: dtypes.NewInt(1_000),
+				},
+				{
+					NumShares: dtypes.NewInt(200),
+				},
+			},
+		},
+		"Two Vaults, One Positive Shares, One Zero Shares": {
+			vaultIds: []vaulttypes.VaultId{
+				constants.Vault_Clob_0,
+				constants.Vault_Clob_1,
+			},
+			totalShares: []vaulttypes.NumShares{
+				{
+					NumShares: dtypes.NewInt(1_000),
+				},
+				{
+					NumShares: dtypes.NewInt(0),
+				},
+			},
+		},
+		"Two Vaults, Both Zero Shares": {
+			vaultIds: []vaulttypes.VaultId{
+				constants.Vault_Clob_0,
+				constants.Vault_Clob_1,
+			},
+			totalShares: []vaulttypes.NumShares{
+				{
+					NumShares: dtypes.NewInt(0),
+				},
+				{
+					NumShares: dtypes.NewInt(0),
+				},
+			},
+		},
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			// Initialize tApp and ctx (in deliverTx mode).
+			tApp := testapp.NewTestAppBuilder(t).WithGenesisDocFn(func() (genesis types.GenesisDoc) {
+				genesis = testapp.DefaultGenesis()
+				// Initialize each vault with quote quantums to be able to place orders.
+				testapp.UpdateGenesisDocWithAppStateForModule(
+					&genesis,
+					func(genesisState *satypes.GenesisState) {
+						subaccounts := make([]satypes.Subaccount, len(tc.vaultIds))
+						for i, vaultId := range tc.vaultIds {
+							subaccounts[i] = satypes.Subaccount{
+								Id: vaultId.ToSubaccountId(),
+								AssetPositions: []*satypes.AssetPosition{
+									{
+										AssetId:  0,
+										Quantums: dtypes.NewInt(1_000_000_000), // 1,000 USDC
+									},
+								},
+							}
+						}
+						genesisState.Subaccounts = subaccounts
+					},
+				)
+				return genesis
+			}).Build()
+			ctx := tApp.InitChain().WithIsCheckTx(false)
+
+			// Set total shares for each vault ID.
+			for i, vaultId := range tc.vaultIds {
+				err := tApp.App.VaultKeeper.SetTotalShares(ctx, vaultId, tc.totalShares[i])
+				require.NoError(t, err)
+			}
+
+			// Check that there's no stateful orders yet.
+			allStatefulOrders := tApp.App.ClobKeeper.GetAllStatefulOrders(ctx)
+			require.Len(t, allStatefulOrders, 0)
```

Comment:
nit: Should we test that this is cancelling existing orders?

Reply:
Good call!